Application/src/view/visuals/component/Message.java

```diff
@@ -157,6 +157,32 @@ public void changeCoordinates(Coordinates node1, Coordinates node2, int class_si
         this.class_size = class_size;
     }
 
+    public void resizeTrail(int oldClassSize){
+         //Checks if there is an existing trail for this message
+
+        //If there is no trail
+        if(trails.size() == 0){
+            return;
+        }
+        //Calculates the new size of the trail image and the difference between the window size
+        double trailsize = class_size/trailScale;
+        double class_size_change = (double)class_size/oldClassSize;
+
+        //Makes sure the trails doesnt move too much to the left
+        int firstX = (int)(trails.get(0).getXcoordinate() * class_size_change);
```

Comment:
try class_size_change-15
looks much better with self calls, and the extra spawned sprite is spawned a bit more to the right of life line which is much more acceptable +1

Reply:
This will just make it so that the sprites go too much to the right when resizing

Reply:
Try -10 instead of -15
Or find a balance so that the self call still looks like a rectangle

Reply:
They look the same as always for me